orchestrator/careplancontributor/handle_taskfiller.go

```diff
@@ -258,6 +265,77 @@ func (s *Service) createSubTaskOrFinishPrimaryTask(cpsClient fhirclient.Client,
 	return nil
 }
 
+// selectWorkflow determines the workflow to use based on the Task's focus, and reasonCode or reasonReference.
+// It first selects the type of service, from the Task.focus (ServiceRequest), and then selects the workflow based on the Task.reasonCode or Task.reasonReference.
+// If it finds no, or multiple, matching workflows, it returns an error.
+func (s *Service) selectWorkflow(cpsClient fhirclient.Client, task *fhir.Task) (*taskengine.Workflow, error) {
+	var matchedServiceCodes []string
+	// Determine service code from Task.focus
+	var serviceRequest fhir.ServiceRequest
+	if err := cpsClient.Read(*task.Focus.Reference, &serviceRequest); err != nil {
```

Comment:
We do not have an explicit endpoint for GET ServiceRequest, won't this fail in test/prod with the most recent change to deny unmanaged operations?

Reply:
Yes, and ServiceRequest (and Condition) are (or should be) on the list to be fixed/authorized. Test environment currently still enables unmanaged FHIR operations.